### PR TITLE
CONTRIBUTING.md: Rewrite from src/CONTRIB

### PR DESCRIPTION
**Pull request by gdt**
This is a wholesale rewrite, prompted by wanting to define which systems are so old that workarounds for them may ve removed.

### CODE REVIEW

**Comment by gdt** [comment]
(This is for post-release; it is nowhere near important enough to derail the rc.)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+need of workarounds.  In practice this means Windows :-)
+
+Be careful not to use GNU/Linux beyond-POSIX interfaces.
+
+Unison has a TUI (text/CLI) which requires very little beyond ocaml.
+There are two GUIs, one with GTK3 and one for mac.  Generally we try
+to be evenhanded with features, but leaving the mac proprietary GUI
+behind is not prohibited.
+
+# Accomodation of old systems
```
Comment (tleedjarv) [inline CONTRIBUTING.md:62]:
```suggestion
# Accommodation of old systems
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Keeping Unison running on old systems can cause accumulation of crufty
+workarounds.  Generally, we aim to keep it workig on even very old
```
Comment (tleedjarv) [inline CONTRIBUTING.md:65]:
```suggestion
workarounds.  Generally, we aim to keep it working on even very old
```

**Comment by tleedjarv** [comment]
Overall looks good to me (I haven't done any final proofreading). Could we merge most of the section "Development and Submitting Proposed Changes" in README.md to this doc here and then refer to this doc from README.md?

**Comment by gdt** [comment]
Good point; I will create a second commit that does the move, and this PR can wait for that -- it's waiting until release anyway.

**Comment by gdt** [comment]
I might squash these three, but will leave it separate today in case anybody finds that more helpful.

File: README.md [hunk 3]
```diff
@@ -10,13 +10,14 @@ before creating or commenting on a github issue.***
 
 ***TL;DR: Do not ask questions or ask for help in issues.  Upgrade to the latest release.***
 
-For compatibility information with version 2.52, see
-https://github.com/bcpierce00/unison/wiki/2.52-Migration-Guide
+If you are trying to interoperate with Unison before 2.52, see the
+compatibility section below.  However, pre-2.52 is now very old; see
+the previous paragraph.
```
Comment (tleedjarv) [inline README.md:15]:
```suggestion
If you are trying to interoperate with or upgrade from Unison before 2.52,
see the compatibility section below.  However, pre-2.52 is now very old;
see the previous paragraph.
```

**Comment by tleedjarv** [comment]
Good to go. Just one small suggestion (feel free to ignore it).

File: README.md (hunk 3 above)
Comment (gdt) [reply README.md]: Good point; edited in.
